libs/ts-rest/core/src/lib/infer-types.ts

```diff
@@ -0,0 +1,128 @@
+import { AppRoute, AppRouteMutation, AppRouter } from './dsl';
+import { HTTPStatusCode } from './status-codes';
+import {
+  Prettify,
+  Without,
+  ZodInferOrType,
+  ZodInputOrType,
+} from './type-utils';
+import { PathParamsWithCustomValidators } from './client';
+
+type AppRouteResponses<
```

Comment:
Do we not want to expose `AppRouteResponses`? In case the developer/consumer would like to play around with it?

Reply:
I think it's a double edged sword. For example exposing something like `ZodInferOrType` allowed developers to build what is essentially this entire PR without having to depend on us adding this functionality, but at the same time it makes backwards compatibility harder for us if we want to do any refactors.

I think a lot of our internal functions and types are being "involuntarily" exposed because we need them to be exported so they can be used internally by other typescript files or other libraries but not because we actually intended to expose them.